### PR TITLE
[Merged by Bors] - feat(Set/Pointwise/SMul): add 2 missing instances

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by eric-wieser** [comment]
@YaelDillies would probably appreciate it if you added the same for `Finset`, if we don't already have them.

**Comment by YaelDillies** [comment]
Yes please!

**Comment by urkud** [comment]
@YaelDillies Could you please have a look at the `Finset` versions?
bors d=YaelDillies

File: Mathlib/Data/Finset/Pointwise.lean [hunk 1]
```diff
@@ -1772,6 +1772,22 @@ scoped[Pointwise]
   attribute [instance]
     Finset.mulActionFinset Finset.addActionFinset Finset.mulAction Finset.addAction
 
+/-- If scalar multiplication by elements of `α` sends `(0 : β)` to zero,
+then the same is true for `(0 : Finset β)`. -/
+protected def smulZeroClassFinset [Zero β] [SMulZeroClass α β] :
+    SMulZeroClass α (Finset β) :=
+  coe_injective.smulZeroClass ⟨(↑), coe_zero⟩ coe_smul_finset
+
+scoped[Pointwise] attribute [instance] Finset.smulZeroClassFinset
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Pointwise.lean:1781]: Can you rather add those instance declarations to line 1805?

Comment (YaelDillies) [reply Mathlib/Data/Finset/Pointwise.lean:1781]: Nevermind, I see it does not play ball with `where` notation.

**Pull request closed without merging**